src/toil/wdl/wdltoil.py

```diff
@@ -1747,40 +1747,71 @@ def run(self, file_store: AbstractFileStore) -> Promised[WDLBindings]:
-                if spec_parts[0] != 'local-disk':
-                    # TODO: Add a WDL line to this error
-                    raise NotImplementedError(f"Could not provide disks = {disks_spec} because only the local-disks mount point is implemented")
-                try:
-                    total_gb += int(spec_parts[1])
-                except:
-                    # TODO: Add a WDL line to this error
-                    raise ValueError(f"Could not parse disks = {disks_spec} because {spec_parts[1]} is not an integer")
+                part_size = None
+                # default to GiB as per spec
+                part_suffix: str = "GiB"
+                # default to the execution directory
+                part_mount_point: str = self._wdl_options.get("execution_dir") or os.getcwd()
+                for i, part in enumerate(spec_parts):
+                    if part.replace(".", "", 1).isdigit():
+                        # round down floats
+                        part_size = int(float(part))
+                        continue
+                    if i == 0:
+                        # mount point is always the first
+                        part_mount_point = part
+                        continue
+                    if part_size is not None:
+                        # suffix will always be after the size, if it exists
+                        part_suffix = part
+                        continue
+
+                if part_size is None:
+                    # Disk spec did not include a size
+                    raise ValueError(f"Could not parse disks = {disks_spec} because {spec} does not specify a disk size")
+
+                per_part_size = convert_units(part_size, part_suffix)
+                total_bytes += per_part_size
+                if mount_spec.get(part_mount_point) is not None:
+                    # raise an error as all mount points must be unique
+                    raise ValueError(f"Could not parse disks = {disks_spec} because the mount point {part_mount_point} is specified multiple times")
+
                 # TODO: we always ignore the disk type and assume we have the right one.
-                # TODO: Cromwell rounds LOCAL disks up to the nearest 375 GB. I
-                # can't imagine that ever being standardized; just leave it
-                # alone so that the workflow doesn't rely on this weird and
-                # likely-to-change Cromwell detail.
-                if spec_parts[2] == 'LOCAL':
+                if part_mount_point != "local-disk":
+                    # Don't mount local-disk. This isn't in the spec, but is carried over from cromwell
+                    mount_spec[part_mount_point] = int(per_part_size)
+
+                if not os.path.exists(part_mount_point):
+                    # this isn't a valid mount point
+                    raise NotImplementedError(f"Cannot use mount point {part_mount_point} as it does not exist")
```

Comment:
This is what the spec says we're supposed to require (the mount point needs to exist on the "host system"), but I've read the disks section of the spec twice and that requirement doesn't really make any sense, because we're mounting storage into the *container*. The spec doesn't say we actually do anything with this path on the host.